website_compassion/controllers/my_account.py

```diff
@@ -153,6 +152,14 @@ def my_children(self, **kwargs):
     def my_child(self, child_id, **kwargs):
         partner = request.env.user.partner_id
         children = _get_user_children()
+        child = children.filtered(lambda c: c.id == child_id)
+        sponsorship_ids = child.sponsorship_ids.filtered(
+            lambda contract: contract.partner_id == partner or
+            contract.correspondent_id == partner
+        )
+        letter_ids = request.env["correspondence"].search([
+            ("sponsorship_id", "=", sponsorship_ids[0].id),
+        ])
```

Comment:
No need to go through children to find the letters.
```suggestion
       letters = request.env["correspondence"].search([
            ("partner_id", "=", partner.id),
            ("child_id", "=", child_id)
        ])
```
I would suggest not to name 'letter_ids' as it holds a recordset and not a list of ids.

Reply:
You're right, I modified the name and will correct in all templates the `_ids` of recordset in the future,